[0.009s] Invoking command in '/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface': CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/install/py_sensor:/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/install/sensor_py:/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/install/py_srvcli:/home/avmi-lab-user/turtlebot4_ws/install/py_sub:/home/avmi-lab-user/turtlebot4_ws/install/path_pubsub:/home/avmi-lab-user/turtlebot4_ws/install/opencv_tests:/home/avmi-lab-user/turtlebot4_ws/install/avmi_lab_threat_field:/opt/ros/humble /usr/bin/cmake --build /home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface -- -j8 -l8
[0.026s] -- Found ament_cmake: 1.3.11 (/opt/ros/humble/share/ament_cmake/cmake)
[0.118s] -- Found geometry_msgs: 4.9.0 (/opt/ros/humble/share/geometry_msgs/cmake)
[0.134s] -- Found rosidl_generator_c: 3.1.6 (/opt/ros/humble/share/rosidl_generator_c/cmake)
[0.136s] -- Found rosidl_adapter: 3.1.6 (/opt/ros/humble/share/rosidl_adapter/cmake)
[0.139s] -- Found rosidl_generator_cpp: 3.1.6 (/opt/ros/humble/share/rosidl_generator_cpp/cmake)
[0.145s] -- Using all available rosidl_typesupport_c: rosidl_typesupport_fastrtps_c;rosidl_typesupport_introspection_c
[0.152s] -- Using all available rosidl_typesupport_cpp: rosidl_typesupport_fastrtps_cpp;rosidl_typesupport_introspection_cpp
[0.175s] -- Found rosidl_default_generators: 1.2.0 (/opt/ros/humble/share/rosidl_default_generators/cmake)
[0.309s] [31mCMake Error at /opt/ros/humble/share/rosidl_adapter/cmake/rosidl_adapt_interfaces.cmake:59 (message):
[0.310s]   execute_process(/usr/bin/python3 -m rosidl_adapter --package-name interface
[0.310s]   --arguments-file
[0.310s]   /home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface/rosidl_adapter__arguments__interface.json
[0.310s]   --output-dir
[0.310s]   /home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface/rosidl_adapter/interface
[0.310s]   --output-file
[0.310s]   /home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface/rosidl_adapter/interface.idls)
[0.310s]   returned error code 1:
[0.310s] 
[0.310s]   Error processing 'float64 Xs' of 'interface/Concentration_Request': ''Xs'
[0.310s]   is an invalid field name.  It should have the pattern
[0.310s]   '^(?!.*__)(?!.*_$)[a-z][a-z0-9_]*$''
[0.310s] 
[0.310s]   Traceback (most recent call last):
[0.310s] 
[0.310s]     File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
[0.310s]       return _run_code(code, main_globals, None,
[0.310s]     File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
[0.310s]       exec(code, run_globals)
[0.310s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/__main__.py", line 19, in <module>
[0.310s]       sys.exit(main())
[0.310s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/main.py", line 53, in main
[0.310s]       abs_idl_file = convert_to_idl(
[0.310s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/__init__.py", line 24, in convert_to_idl
[0.310s]       return convert_srv_to_idl(
[0.310s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/srv/__init__.py", line 28, in convert_srv_to_idl
[0.311s]       srv = parse_service_string(package_name, input_file.stem, content)
[0.311s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/parser.py", line 859, in parse_service_string
[0.311s]       request_message = parse_message_string(
[0.311s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/parser.py", line 519, in parse_message_string
[0.311s]       fields.append(Field(
[0.311s]     File "/opt/ros/humble/local/lib/python3.10/dist-packages/rosidl_adapter/parser.py", line 354, in __init__
[0.311s]       raise NameError(
[0.311s] 
[0.311s]   NameError: 'Xs' is an invalid field name.  It should have the pattern
[0.311s]   '^(?!.*__)(?!.*_$)[a-z][a-z0-9_]*$'
[0.311s] 
[0.311s] Call Stack (most recent call first):
[0.311s]   /opt/ros/humble/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:130 (rosidl_adapt_interfaces)
[0.311s]   CMakeLists.txt:17 (rosidl_generate_interfaces)
[0.311s] 
[0.311s] [0m
[0.311s] -- Configuring incomplete, errors occurred!
[0.311s] See also "/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface/CMakeFiles/CMakeOutput.log".
[0.314s] gmake: *** [Makefile:1179: cmake_check_build_system] Error 1
[0.316s] Invoked command in '/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface' returned '2': CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/install/py_sensor:/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/install/sensor_py:/home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/install/py_srvcli:/home/avmi-lab-user/turtlebot4_ws/install/py_sub:/home/avmi-lab-user/turtlebot4_ws/install/path_pubsub:/home/avmi-lab-user/turtlebot4_ws/install/opencv_tests:/home/avmi-lab-user/turtlebot4_ws/install/avmi_lab_threat_field:/opt/ros/humble /usr/bin/cmake --build /home/avmi-lab-user/ThreatField2025/workspace/ros2_ws/build/interface -- -j8 -l8
